src/test/java/com/spotify/docker/client/DefaultDockerClientTest.java

```diff
@@ -455,7 +455,8 @@ public void testPullPrivateRepoWithoutAuth() throws Exception {
   public void testBuildImageIdWithBuildargs() throws Exception {
     requireDockerApiVersionAtLeast("1.21", "build args");
 
-    final String dockerDirectory = Resources.getResource("dockerDirectoryWithBuildargs").getPath();
+    final String dockerDirectory = 
+        Paths.get(Resources.getResource("dockerDirectoryWithBuildargs").toURI()).toString();
```

Comment:
this line is repeated many times but with different directory names - can you wrap it in a method and also have a comment in the method explaining why the URL -> URI -> String -> Path dance is needed?